File: src/creational/abstract-factory/typescript/classes/EnterpriseVehicle.ts
import { VehicleProtocol } from './interfaces/VehicleProtocol';

export class EnterpriseVehicle extends VehicleProtocol {
  private enterpriser: string | null = null;

  pickUp(userName: string): void {
    this.enterpriser = userName;
    console.log(`${this.name} is looking for ${userName}`);
  }

  get user(): string | null {
    return this.enterpriser;
  }
}

File: src/creational/abstract-factory/python/classes/enterprise_vehicle.py
from classes.interfaces.vehicle_protocol import VehicleProtocol


class EnterpriseVehicle(VehicleProtocol):
    def pick_up(self, user_name: str) -> None:
        self.enterpriser = user_name
        print(f'{self.get_name()} is looking for {user_name}')

    def get_user(self):
        return self.enterpriser if self.enterpriser else None
